### PR TITLE
[EDIFICE] Champ positions

### PR DESCRIPTION
**Pull request by juniorode**
# Description

Please include a summary of the changes and the related issue.

## Fixes

(Enter here Jira or Redmine ticket(s) links)

## Type of change

Please check options that are relevant.

- [ ] Chore (PATCH)
- [ ] Doc (PATCH)
- [ ] Bug fix (PATCH)
- [ ] New feature (MINOR)

## Which packages changed?

Please check the name of the package you changed

- [ ] admin
- [ ] app-registry
- [ ] archive
- [ ] auth
- [ ] cas
- [ ] common
- [ ] communication
- [ ] conversation
- [ ] directory
- [ ] feeder
- [ ] infra
- [ ] portal
- [ ] session
- [ ] test
- [ ] tests
- [ ] timeline
- [ ] workspace

## Tests

`docker compose run --rm k6 run file:///home/k6/src/it/scenarios/position/crud.js`
`docker compose run --rm k6 run file:///home/k6/src/it/scenarios/position/attribute-position.js`

# Reminder

- Security flaws
- Performance impacts (think bulk !)
- Unit tests were replayed
- Unit tests were added and/or changed
- I have updated the reminder for the version including my modifications

- [ ] All done ! :smiley:

### CODE REVIEW

File: tests/src/test/js/it/scenarios/position/crud.js [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    assertKo(res, "An unauthenticated user should not be able to create a position");
+
+    session = authenticateWeb(teacher.login)
+    res = createPosition(positionName, structure, session);
+    assertKo(res, "An authenticated user without special rights should not be able to create a position");
+
+
+    session = authenticateWeb(adml.login)
+    res = createPosition(positionName, structure, session);
+    assertKo(res, "An ADML user should not be able to create a position");
```
[the remark this answers is not in the document]
Comment (mariusode) [reply tests/src/test/js/it/scenarios/position/crud.js:136]: Peut-être à transformer du coup en : 
```
    // An ADML should be able to create a position in his structure
    session = authenticateWeb(adml1.login)
    createPositionOrFail(positionName, structure1, session);
```

File: tests/src/test/js/it/scenarios/position/crud.js [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+    assertKo(res, "An authenticated user without special rights should not be able to search positions");
+
+
+    session = authenticateWeb(asml.login)
+    res = searchPositions('Coucou', session);
+    assertOk(res, "An ADML should be able to search positions");
+
+
+    session = authenticateWeb(__ENV.ADMC_LOGIN, __ENV.ADMC_PASSWORD);
+    res = searchPositions('Coucou', structure, session);
```
Comment (mariusode) [inline tests/src/test/js/it/scenarios/position/crud.js:182]: Missing 
`    assertOk(res, "An ADMC should be able to search positions");`

File: tests/src/test/js/it/scenarios/position/crud.js [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+
+    let res = searchPositions('MEtab1', session);
+    check(JSON.parse(res.body), {
+      'adml of structure1 should see the positions of structure 1 only': pos => pos && pos.length === positions1.length,
+      'all positions of structure1 should be fetched': pos => allPositionsOk(positions1, pos),
+      'no duplicates were returned': pos =>   (positions1)
+    })
+
+    session = authenticateWeb(adml2.login)
+    res = searchPositions('MEtab2', session);
```
Comment (mariusode) [inline tests/src/test/js/it/scenarios/position/crud.js:222]: idem

File: tests/src/test/js/it/scenarios/position/crud.js [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+ * @param {*} param0 initial data 
+ */
+export function testDeletePosition({structureTree}) {
+  const {structures: [structure1, structure2], admls: [adml1], headAdml} = structureTree
+  describe("[Position-CRUD] Delete position", () => {
+    const session = authenticateWeb(__ENV.ADMC_LOGIN, __ENV.ADMC_PASSWORD);
+
+    ////////////////////////////////////
+    // Create 2 positions for structure
+    // 1 and 1 position for structure 2
```
Comment (mariusode) [inline tests/src/test/js/it/scenarios/position/crud.js:253]: and 3* positions for structure 2

File: tests/src/test/js/it/scenarios/position/attribute-position.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    const megaAdml = getRandomUserWithProfile(users2, 'Teacher', [adml2]);
+    makeAdml(megaAdml, chapeau, session)
+    ////////////////////////////////////
+    // Create 1 position for each structure
+    const position1 = createPositionOrFail(`IT - Position 1`, structure1, session);
+    const position2 = createPositionOrFail(`IT - Position 2`, structure2, session);
+    structureTree = { 
+        head: chapeau, 
+        structures: [structure1, structure2], 
+        admls: [structure1, structure2],
```
[the remark this answers is not in the document]
Comment (juniorode) [reply tests/src/test/js/it/scenarios/position/attribute-position.js]: Fait

File: tests/src/test/js/it/scenarios/position/crud.js [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+    // and 1 ADML for the head structure
+    const users1 = getUsersOfSchool(structure1, session)
+    const adml1 = getRandomUserWithProfile(users1, 'Teacher');
+    makeADML(adml1, structure1, session)
+    const users2 = getUsersOfSchool(structure2, session)
+    const adml2 = getRandomUserWithProfile(users2, 'Teacher');
+    makeADML(adml2, structure2, session)
+    const megaAdml = getRandomUserWithProfile(users2, 'Teacher', [adml2]);
+    makeADML(megaAdml, structure, session)
+    structureTree = { head: chapeau, structures: [structure1, structure2], admls: [structure1, structure2], headAdml: megaAdml}
```
[the remark this answers is not in the document]
Comment (juniorode) [reply tests/src/test/js/it/scenarios/position/crud.js]: Fait

File: tests/src/test/js/it/scenarios/position/crud.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (juniorode) [reply tests/src/test/js/it/scenarios/position/crud.js]: Fait

File: tests/src/test/js/it/scenarios/position/crud.js [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+    let res = searchPositions('Coucou');
+    assertKo(res, "An unauthenticated user should not be able to search positions");
+
+
+    session = authenticateWeb(teacher.login)
+    res = searchPositions('Coucou', session);
+    assertKo(res, "An authenticated user without special rights should not be able to search positions");
+
+
+    session = authenticateWeb(asml.login)
```
[the remark this answers is not in the document]
Comment (juniorode) [reply tests/src/test/js/it/scenarios/position/crud.js]: Fait

File: tests/src/test/js/it/scenarios/position/crud.js (hunk 2 above)
Comment (juniorode) [reply tests/src/test/js/it/scenarios/position/crud.js:182]: Fait

File: tests/src/test/js/it/scenarios/position/crud.js [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+    ]
+    const positions2 = [
+      createPositionOrFail(`IT Position - Search - MEtab2 Coucou`, structure2, session),
+      createPositionOrFail(`IT Position - Search - MEtab2 Hello`, structure2, session),
+      createPositionOrFail(`IT Position - Search - MEtab2 Hello les amis`, structure2, session)
+    ]
+
+    session = authenticateWeb(adml1.login)
+
+    let res = searchPositions('MEtab1', session);
```
[the remark this answers is not in the document]
Comment (juniorode) [reply tests/src/test/js/it/scenarios/position/crud.js]: Fait

File: tests/src/test/js/it/scenarios/position/crud.js (hunk 4 above)
Comment (juniorode) [reply tests/src/test/js/it/scenarios/position/crud.js]: Fait

**Comment by jcbe-ode** [comment]
Squash pour première livraison sur PF de recette.

**Pull request merged**